File: Phidget Code/src/Models/In/Stand.java
package Models.In;

import Events.StandEvent;
import Models.BlackBox.BlackBox;
import com.phidgets.InterfaceKitPhidget;
import com.phidgets.PhidgetException;
import com.phidgets.event.InputChangeEvent;
import com.phidgets.event.InputChangeListener;

/**
 * Created by bri_e on 08-03-17.
 */
public class Stand {

    private InterfaceKitPhidget interfaceKitPhidget;
    private BlackBox blackBox;
    private String standName;
    private boolean[] seats;
    private int numberOfSeats;


    public Stand(InterfaceKitPhidget ifk, String name, int nbrSeats,BlackBox bbx){
        standName = name;
        interfaceKitPhidget = ifk;
        blackBox = bbx;
        numberOfSeats = nbrSeats;
        seats = new boolean[numberOfSeats];
        interfaceKitPhidget.addInputChangeListener(new InputChangeListener() {
            @Override
            public void inputChanged(InputChangeEvent inputChangeEvent) {
                if (inputChangeEvent.getState()) {
                    int seatOfConcern = inputChangeEvent.getIndex();
                    if (seats[seatOfConcern]) {
                        seats[seatOfConcern] = false;
                        System.out.println("Siège n°" + seatOfConcern + " libéré dans la tribune " + standName);
                    } else {
                        seats[seatOfConcern] = true;
                        System.out.println("Siège n°" + seatOfConcern + " occupé dans la tribune " + standName);
                    }
                    try {
                        blackBox.processElement(new StandEvent(System.currentTimeMillis()));
                    } catch(PhidgetException e) {
                        System.out.println("Error while processing event");
                    }
                }
            }
        });
    }

    public int getNumberOfSeats() {
        return numberOfSeats;
    }

    public String getStandName() {
        return standName;
    }

    public boolean[] getSeats() {
        return seats;
    }
}

File: PhidgetScalaCode/src/Models/In/Stand.scala
package Models.In

import BlackBox.BlackBox
import Events.StandEvent
import com.phidgets.{InterfaceKitPhidget, PhidgetException}
import com.phidgets.event.{InputChangeEvent, InputChangeListener}

/**
  * Created by bri_e on 20-04-17.
  * A stand has a certain number of seats which are taken or not
  */
class Stand(interfaceKitPhidget: InterfaceKitPhidget, standName: String, numberOfSeats: Int, blackBox: BlackBox) {

  var seats:Array[Boolean] = new Array[Boolean](numberOfSeats)

  interfaceKitPhidget.addInputChangeListener((inputChangeEvent: InputChangeEvent) => {
    if (inputChangeEvent.getState) {
      val seatOfConcern = inputChangeEvent.getIndex
      if (seats(seatOfConcern)) {
        seats(seatOfConcern) = false
        println("Siège n°" + seatOfConcern + " libéré dans la tribune " + standName)
      }
      else {
        seats(seatOfConcern) = true
        println("Siège n°" + seatOfConcern + " occupé dans la tribune " + standName)
      }
      try
      //Todo: Process element || blackBox.processElement(new StandEvent(System.currentTimeMillis))
      catch {
        case e: PhidgetException =>
          println("Error while processing event")
      }
    }
  })

  def getNumberOfSeats: Int = numberOfSeats

  def getStandName: String = standName

  def getSeats: Array[Boolean] = seats
}
